### PR TITLE
padding/enlargement improvements

### PR DESCRIPTION
**Pull request by behinger**
- allow pad_value to be a function (e.g. mean)
- rename protected 'padding' word to `enlarge`

Helps to investigate #9, also `padding=0` did not work prior

### CODE REVIEW

File: src/core-recipe.jl [hunk 1]
```diff
@@ -87,8 +87,12 @@ function Makie.plot!(p::TopoPlot)
         return
     end
     notify(p.resolution) # trigger above (we really need `update=true` for onany)
-
-    padded_data = lift(pad_data, p.data, npositions, p.pad_value)
+    
+    if !isempty(methods(to_value(p.enlarge_value)))
```
Comment (palday) [inline src/core-recipe.jl:91]: calling `methods` explicitly is a bit weird. What do you want to do here?

Comment (behinger) [reply src/core-recipe.jl:91]: find out whether elarge_value is a function or not. This is what I googled how to find out whether a variable is callable or not. I guess, one could also check whether it is a Number, and else call the function. probably easier to understand

**Comment by palday** [comment]
Seems to be superseded by #20. If there is still something good here, then we need to merge `master` in this branch after #20 lands.

**Comment by SimonDanisch** [comment]
I don't think there is...If we want to use the mean for extrapolation, we'd need a new PR with a new extrapolator! ;)

**Pull request closed without merging**